### PR TITLE
DCS--1516 Support confirmation of unexpected arrivals

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: helm_deploy/values-dev.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ generic-service:
     TOKEN_VERIFICATION_API_URL: "https://token-verification-api-dev.prison.service.justice.gov.uk"
     WELCOME_API_URL: "https://welcome-api-dev.prison.service.justice.gov.uk"
     HMPPS_COOKIE_NAME: hmpps-session-dev
-    HMPPS_COOKIE_DOMAIN: welcome-dev.prison.service.justice.gov.uk
+    HOSTNAME: welcome-dev.prison.service.justice.gov.uk
```
Comment (andrewrlee) [inline helm_deploy/values-dev.yaml:19]: renamed this as it is now more generally used than just for cookies

File: server/routes/bookedtoday/arrivals/autoMatchingRecords/multipleMatchingRecordsFoundController.test.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { expectSettingCookie } from '../../../__testutils/requestTestUtils'
 import { State } from '../state'
 
 jest.mock('../../../../services/expectedArrivalsService')
-const expectedArrivalsService = new ExpectedArrivalsService(null, null) as jest.Mocked<ExpectedArrivalsService>
+const expectedArrivalsService = new ExpectedArrivalsService(null, null, null) as jest.Mocked<ExpectedArrivalsService>
```
Comment (andrewrlee) [inline server/routes/bookedtoday/arrivals/autoMatchingRecords/multipleMatchingRecordsFoundController.test.ts:15]: have to add new parameter everywhere as service is now responsible for raising GA events

File: server/routes/bookedtoday/arrivals/confirmArrival/checkAnswersController.ts [hunk 3]
```diff
@@ -29,33 +29,13 @@ export default class CheckAnswersController {
       const { id } = req.params
       const { username, activeCaseLoadId } = res.locals.user
       const arrival = State.newArrival.get(req)
-      const data = await this.expectedArrivalsService.getArrival(id)
```
Comment (andrewrlee) [inline server/routes/bookedtoday/arrivals/confirmArrival/checkAnswersController.ts:32]: pushed this logic down into service

File: server/services/expectedArrivalsService.ts [hunk 4]
```diff
@@ -67,12 +72,53 @@ export default class ExpectedArrivalsService {
...
+        'Confirmed arrival',
+        `AgencyId: ${prisonId}, From: ${data.fromLocation}, Type: ${data.fromLocationType},`,
+        config.hostname
+      )
+    } else {
+      this.raiseAnalyticsEvent(
+        'Add to the establishment roll',
+        'Confirmed unexpected arrival',
+        `AgencyId: ${prisonId}`,
+        config.hostname
```
Comment (andrewrlee) [inline server/services/expectedArrivalsService.ts:117]: this should be pushed into the function - will do as a follow up

**Review by louisenorris — APPROVED** [review]
LGTM

File: helm_deploy/values-dev.yaml (hunk 1 above)
Comment (andrewrlee) [reply helm_deploy/values-dev.yaml]: seemed like HOSTNAME wasn't a valid name so went with APP_HOSTNAME

File: server/services/expectedArrivalsService.ts (hunk 4 above)
Comment (andrewrlee) [reply server/services/expectedArrivalsService.ts]: done this now

**Review by louisenorris — APPROVED** [review]
LGTM

**Pull request merged**